### ISSUE TITLE
jax.jit decorator issue for the calculate_distance() method in Structure class

### ISSUE THREAD

**Issue by hghcomphys** [hghcomphys/jaxip#62]
It raises an error when calling `calculate_distance()` after JIT compiling a structure instance for the second time in the notebooks.
It complains about a bool value and any()/all() ambiguous statement.

**Comment by hghcomphys** [comment]
It seemed that the issue was related to calling `calculate_distance()` inside the neighbor's cutoff mask kernel function. I used the _calculate_distance kernel (jit) and the issue disappeared.
